hs_core/management/commands/repair_resource.py

```diff
@@ -25,47 +25,100 @@ class Command(BaseCommand):
+            cuttoff_time = timezone.now() - timedelta(days)
+            resources = resources.filter(updated__gte=cuttoff_time)
+
+        if dry_run:
+            print("CONDUCTING A DRY RUN: FIXES WILL NOT BE SAVED")
+
+        if not resources:
+            print("NO RESOURCES FOUND MATCHING YOUR FILTER ARGUMENTS")
+            return
+
+        resources = resources.order_by(F('updated').asc(nulls_first=True))
+
+        total_res_to_check = resources.count()
+        impacted_resources = 0
+        total_files_missing_in_django = 0
+        total_files_dangling_in_django = 0
+        resources_with_missing_django = []
+        resources_with_missing_irods = []
+        for count, resource in enumerate(resources):
+            res_url = site_url + resource.get_absolute_url()
+            print("*" * 100)
+            print(f"{count}/{total_res_to_check}: Checking resource {res_url}")
+            _, missing_in_django, dangling_in_django = repair_resource(resource, logger, dry_run=dry_run)
+            if dangling_in_django > 0 or missing_in_django > 0:
+                impacted_resources += 1
+                total_files_missing_in_django += missing_in_django
+                total_files_dangling_in_django += dangling_in_django
+                if missing_in_django > 0:
+                    resources_with_missing_django.append(res_url)
+                if dangling_in_django > 0:
+                    resources_with_missing_irods.append(res_url)
+                print(f"{dangling_in_django} files dangling in Django for this resource.")
+                print(f"{missing_in_django} files missing in Django for this resource.")
+                print(f"Resources thus far with at least one missing django file: {len(resources_with_missing_django)}")
+                print(f"Resources thus far with at least one dangling django file: {len(resources_with_missing_irods)}")
+                print(f"Total resources with discrepancies thus far: {impacted_resources}")
+        print("*" * 100)
+        print("*" * 100)
+        print(f"Number of resources that had at least one file issue: {impacted_resources}")
+
+        print("*" * 100)
+        print(f"Total number of files missing in Django (across all checked resources): \
+              {total_files_missing_in_django}")
+        print(f"Number of resources with at least one missing django file: {len(resources_with_missing_django)}")
+        for res in resources_with_missing_django:
+            print(res)
+
+        print("*" * 100)
+        print(f"Total number of files dangling in Django (across all checked resources): \
+              {total_files_dangling_in_django}")
+        print(f"Number of resources with at least one missing irods file: {len(resources_with_missing_irods)}")
+        for res in resources_with_missing_irods:
+            print(res)
+
+        # Make it simple to detect clean/fail run in Jenkins
+        if impacted_resources:
```

Comment:
`if not impacted_resources:`?

Reply:
My intent is to raise an exception (which can easily be caught in Jenkins) when there are resources that have disparities. In other words, `if impacted_resources` ... so I think this should stay as is.